youtube_dl/extractor/reddit.py

```diff
@@ -47,7 +47,7 @@ def _real_extract(self, url):
 
 
 class RedditRIE(InfoExtractor):
-    _VALID_URL = r'(?P<url>https?://(?:(?:www|old)\.)?reddit\.com/r/[^/]+/comments/(?P<id>[^/?#&]+))'
+    _VALID_URL = r'(?P<url>https?://(?:(?:.*)\.)?reddit\.com/r/[^/]+/comments/(?P<id>[^/?#&]+))'
```

Comment:
Should not allow empty 3rd level domain. Should not be greedy. Inner group is superfluous.

Reply:
Should it match every 2 or 3 alphabet 3rd level domain then? All reddit subdomains are either 2 alphabet (e.g. fr, de, en) or 3 alphabet (e.g. www, old).

Reply:
It should match all non empty domain names.